packages/nvidia_nat_redis/tests/test_redis_object_store.py

```diff
@@ -0,0 +1,52 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import socket
+from contextlib import asynccontextmanager
+
+import pytest
+
+from nat.builder.workflow_builder import WorkflowBuilder
+from nat.plugins.redis.object_store import RedisObjectStoreClientConfig
+from nat.test.object_store_tests import ObjectStoreTests
+
+# NOTE: This test requires a local Redis server to be running.
+# To launch a local server using docker, run the following command:
+# docker run --rm -ti --name test-redis -p 6379:6379 redis:7-alpine
+
+
+def _redis_available(host: str = "localhost", port: int = 6379) -> bool:
+    with socket.socket() as s:
+        s.settimeout(0.25)
+        try:
+            s.connect((host, port))
+            return True
+        except OSError:
+            return False
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(not _redis_available(), reason="Redis server not available")
```

Comment:
I think this would work better as a fixture, this would allow forcing a fail rather than a skip if someone used the `--fail_missing` flag.